bitwarden_license/src/Commercial.Core/SecretManagerFeatures/ServiceAccounts/UpdateServiceAccountCommand.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.SecretManagerFeatures.ServiceAccounts.Interfaces;
+
+namespace Bit.Commercial.Core.SecretManagerFeatures.ServiceAccounts;
+
+public class UpdateServiceAccountCommand : IUpdateServiceAccountCommand
+{
+    private readonly IServiceAccountRepository _serviceAccountRepository;
+
+    public UpdateServiceAccountCommand(IServiceAccountRepository serviceAccountRepository)
+    {
+        _serviceAccountRepository = serviceAccountRepository;
+    }
+
+    public async Task<ServiceAccount> UpdateAsync(ServiceAccount serviceAccount)
+    {
+        if (serviceAccount.Id == default(Guid))
+        {
+            throw new BadRequestException("Cannot update service account, service account does not exist.");
```

Comment:
Why are we returning a different response for this compared to below?

Reply:
Mainly to stay consistent with the other secret manager update commands. 

When I wrote the Secrets update, we had the ID included in the request body. 

If the client left the ID blank, a default GUID would be generated. 

With how things are structured now, it would probably be better to just take these checks out of the update commands.

What are your thoughts?

Reply:
Please remove it.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2323/commits/aba0e0b4e4c398069beb462724381d36b5b43773